src/remove_intersection.cpp

```diff
@@ -0,0 +1,172 @@
+#include "remove_intersection.hpp"
+
+// stlib
+#include <vector>
+#include <algorithm>
+
+
+Texture RemoveIntersection::spacebar_texture;
+RemoveIntersection::RemoveIntersection() :
+	m_press(0)
+{
+
+}
+
+RemoveIntersection::~RemoveIntersection()
+{
+
+}
+bool RemoveIntersection::init()
+{
+	show = false;
+	curr_frame = 0;
+	m_prev_time = 0.f;
+	m_fps = 300.f;
+
+
+	// Load shared texture
+	if (!spacebar_texture.is_valid())
+	{
+		if (!spacebar_texture.load_from_file(textures_path("spacebar1.png")))
+		{
+			fprintf(stderr, "Failed to load cop texture!");
+			return false;
+		}
+	}
+
+	// The position (0,0) corresponds to the center of the texture
+	float wr = spacebar_texture.width * 0.5;
+	float hr = spacebar_texture.height * 0.5;
+
+	TexturedVertex vertices[4];
+	vertices[0].position = { -wr, +hr, 0.f };
+	vertices[0].texcoord = { 0.f, 1.f };//top left
+	vertices[1].position = { +wr, +hr, 0.f };
+	vertices[1].texcoord = { 1.f, 1.f };//top right
+	vertices[2].position = { +wr, -hr, 0.f };
+	vertices[2].texcoord = { 1.f, 0.f };//bottom right
+	vertices[3].position = { -wr, -hr, 0.f };
+	vertices[3].texcoord = { 0.f, 0.f };//bottom left
+
+										// counterclockwise as it's the default opengl front winding direction
+	uint16_t indices[] = { 0, 3, 1, 1, 3, 2 };
+
+	// Clearing errors
+	gl_flush_errors();
+
+	// Vertex Buffer creation
+	glGenBuffers(1, &mesh.vbo);
+	glBindBuffer(GL_ARRAY_BUFFER, mesh.vbo);
+	glBufferData(GL_ARRAY_BUFFER, sizeof(TexturedVertex) * 4, vertices, GL_STATIC_DRAW);
+
+	// Index Buffer creation
+	glGenBuffers(1, &mesh.ibo);
+	glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, mesh.ibo);
+	glBufferData(GL_ELEMENT_ARRAY_BUFFER, sizeof(uint16_t) * 6, indices, GL_STATIC_DRAW);
+
+	// Vertex Array (Container for Vertex + Index buffer)
+	glGenVertexArrays(1, &mesh.vao);
+	if (gl_has_errors())
+		return false;
+
+	// Loading shaders
+	if (!effect.load_from_file(shader_path("textured.vs.glsl"), shader_path("textured.fs.glsl")))
+		return false;
+
+	// Setting initial values, scale is negative to make it face the opposite way
+	// 1.0 would be as big as the original texture
+	m_scale.x = 0.5;
+	m_scale.y = 0.5;
+	m_position = { 1000.f, 850.f };
+	//m_rotation = 0.f;
+
+	return true;
+}
+
+void RemoveIntersection::update(float ms, int hit_count)
+{
+	m_prev_time += ms;
+	if (m_prev_time >=0 && m_prev_time < 150) {
+		curr_frame = (curr_frame + 1) % 6;
+		spacebar_texture.load_from_file(textures_path("spacebar1.png"));
```

Comment:
This line and line 95 is partly the reason for ramping process memory. Perhaps look into using glActiveTexture, or load those two textures in the init, and then use the textures from there rather than loading and using the texture on each call.